javatests/arcs/core/host/ParticleRegistrationTest.kt

```diff
@@ -0,0 +1,88 @@
+package arcs.core.host
+
+import arcs.core.sdk.Particle
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class ParticleRegistrationTest {
+    @Test
+    fun serviceLoader_allParticlesAreRegistered() {
+        var foundProdHost = false
+        var foundTestHost = false
+
+        ServiceLoaderHostRegistry.instance.availableArcHosts().forEach { host: ArcHost ->
+            when (host) {
+                is ProdHost -> {
+                    assertThat(TestProdParticle::class).isIn(
+                        host.registeredParticles()
+                    )
```

Comment:
Nit: for this, and the other places you're testing that something is/isn't in a collection, it's best to switch the order, as the semantics of the assertion errors thrown by Truth rely on the following format:

```kotlin
assertThat(actual).isSomething(expected)
```

So this would become:

```kotlin
assertThat(host.registeredParticles()).contains(TestProdParticle::class)
```

Reply:
Done